frappe/public/js/frappe/form/controls/duration.js

```diff
@@ -0,0 +1,145 @@
+frappe.ui.form.ControlDuration = frappe.ui.form.ControlData.extend({
+	make_input: function() {
+		this._super();
+		this.make_picker();
+	},
+
+	make_picker: function() {
+		this.inputs = [];
+		this.set_duration_options();
+		this.$picker = $(
+			`<div class="duration-picker">
+				<div class="picker-row row"></div>
+			</div>`
+		);
+		this.$wrapper.append(this.$picker);
+		this.build_numeric_input("days", !this.duration_options.showDays);
+		this.build_numeric_input("hrs", false);
+		this.build_numeric_input("mins", false);
+		this.build_numeric_input("secs", !this.duration_options.showSeconds);
+		this.set_duration_picker();
+		this.$picker.hide();
+		this.bind_events();
+		this.refresh();
+	},
+
+	build_numeric_input: function(label, hidden, max) {
+		let $duration_input = $(`
+			<input class="input-sm duration-input" data-duration="${label}" type="number" min="0" value="0">
+		`)
+
+		let $input = $(`<div class="row duration-row"></div>`).prepend($duration_input);
+
+		if (max) {
+			$duration_input.attr("max", max);
+		}
+
+		this.inputs[label] = $duration_input;
+
+		let $control = $(`
+			<div class="col duration-col">
+				<div class="row duration-row duration-label">${label}</div>
+			</div>`
+		)
+
+		if (hidden) {
+			$control.addClass("hidden");
+		}
+		$control.prepend($input);
+		$control.appendTo(this.$picker.find(".picker-row"));
+	},
+
+	set_duration_options() {
+		this.duration_options = frappe.meta.get_duration_options(this.df, this.get_doc());
+	},
+
+	set_duration_picker() {
+		let total_duration = frappe.utils.seconds_to_duration(this.value, this.duration_options);
+
+		if (this.$picker) {
+			if (total_duration.days) {
+				this.$picker.find(`[data-duration="days"]`).prop("value", total_duration.days);
+			}
+			if (total_duration.hours) {
+				this.$picker.find(`[data-duration="hrs"]`).prop("value", total_duration.hours);
+			}
+			if (total_duration.minutes) {
+				this.$picker.find(`[data-duration="mins"]`).prop("value", total_duration.minutes);
+			}
+			if (total_duration.seconds) {
+				this.$picker.find(`[data-duration="secs"]`).prop("value", total_duration.seconds);
+			}
+		}
+	},
+
+	bind_events: function() {
+		let me = this;
+		let clicked = false;
+
+		this.$picker.on("change", ".duration-input", () => {
+			clicked = false;
+			me.set_value(me.duration_to_seconds());
+			me.set_focus();
+		});
+
+		this.$wrapper.find(".duration-input").mousedown(() => {
```

Comment:
Why such an event and the `clicked` flag?

Reply:
The `clicked` flag is used for deciding when to close the picker. The picker should be closed on blur event only when the element causing the blur event is not among the individual input boxes (days, hours, minutes, seconds). The `mousedown` event here means the duration is being set in one of the input boxes, the picker should not be closed in this case hence, `clicked` is set to true. It is used in place of `click` event to keep the picker visible from `mousedown` to change and `clicked` is set back to false on change event. Added comments for better understanding.

P.S.: The PR was not ready for review. Sorry, I should have closed it.

Reply:
Can you add this in code comments, so that anyone looking at it in the future is not confused